tools/importer-msgraph-metadata/pipeline/parser_resources.go

```diff
@@ -0,0 +1,98 @@
+package pipeline
+
+import (
+	"net/http"
+)
+
+type Resource struct {
+	Name       string
+	Category   string
+	Version    string
+	Service    string
+	Paths      []ResourceId
+	Operations []Operation
+}
+
+type Operation struct {
+	Name         string
+	Type         OperationType
+	Method       string
+	ResourceId   *ResourceId
+	UriSuffix    *string
+	RequestModel *string
+	Responses    Responses
+	Tags         []string
+}
+
+type Response struct {
+	Status      int
+	ContentType *string
+	Collection  bool
+	ModelName   *string
+}
+
+type Responses []Response
+
+func (rs Responses) FindModelName() string {
+	for _, r := range rs {
+		if r.ModelName != nil {
+			return *r.ModelName
+		}
+	}
+	return ""
+}
+
+type OperationType uint8
+
+const (
+	OperationTypeUnknown OperationType = iota
+	OperationTypeList
+	OperationTypeRead
+	OperationTypeCreate
+	OperationTypeCreateUpdate
+	OperationTypeUpdate
+	OperationTypeDelete
+)
+
+func NewOperationType(method string) OperationType {
+	switch method {
+	case http.MethodGet:
+		return OperationTypeRead
+	case http.MethodPost:
+		return OperationTypeCreate
+	case http.MethodPatch:
+		return OperationTypeUpdate
+	case http.MethodPut:
+		return OperationTypeCreateUpdate
+	case http.MethodDelete:
+		return OperationTypeDelete
+	}
+	return OperationTypeUnknown
+}
+
+type Resources map[string]*Resource
+
+func (r Resources) ServiceHasValidResources(serviceName string) bool {
+	for _, resource := range r {
+		if resource.Category == "" {
+			continue // TODO do something about orphaned resources
```

Comment:
It used to be a lot, but with recent improvements it's actually only 2 paths now (`/places/$count` and `/places/graph.room/$count`), which is definitely an edge case since (a) there's no `/places` endpoint and (b) the longer of the two is an odata function against a type cast (!!). Both should actually probably be ignored right now, but I guess we need a way to surface these so I'll just log it for now.